tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -11312,10 +11310,19 @@ void DiagnoseEntryAttrAllowedOnStage(clang::Sema *self,
       switch (pAttr->getKind()) {
 
       case clang::attr::HLSLWaveSize: {
-        bool isValid = llvm::StringSwitch<bool>(stageName)
-                           .Case("compute", true)
-                           .Case("node", true)
-                           .Default(false);
+        switch (shaderKind) {
+        case DXIL::ShaderKind::Compute: {
+          isValid = true;
+          break;
+        }
+        case DXIL::ShaderKind::Node: {
```

Comment:
Why is this so much more code than necessary?  Why not just:
```cpp
switch (shaderKind) {
case DXIL::ShaderKind::Compute:
case DXIL::ShaderKind::Node:
  break;
default:
  self->Diag(pAttr->getRange().getBegin(),
                     diag::err_hlsl_attribute_unsupported_stage)
              << "WaveSize"
              << "compute or node";
  break;
}
```
Plus the `isValid` variable is defined and initialized outside the attribute loop, meaning it will carry a stale value around the loop which could be an issue if there's a path that doesn't update the flag for some attribute in the future.  I don't think it's needed at all.